annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,177 @@
+        restoreJsonParamsBuilder.set("gigawordDataDirectory", params.getString("gigawordDataDirectory"))
+        restoreJsonParamsBuilder.set("inputJsonDirectory", params.getString("inputJsonDirectory"))
+        restoreJsonParamsBuilder.set("restoredJsonDirectory", params.getString("restoredJsonDirectory"))
+    }
+    val restoreJsonParams = restoreJsonParamsBuilder.build()
+
+    val repoToPushTo = params.getString("repoToPushTo")
+    val localWorkingCopyDirectory = params.getString("localWorkingCopyDirectory")
+
+    val git = setUpRepository(localWorkingCopyDirectory, repoToPushTo)
+
+    // Run ExportAnnotations.kt
+    pushLogger.info {"Beginning ExportAnnotations"}
+    ExportAnnotations.export(exportAnnotationsParams)
+
+    // Run RestoreJson.kt
+    if (restoreJson) {
+        pushLogger.info { "Beginning RestoreJson" }
+        RestoreJson.restore(restoreJsonParams)
+    }
+
+    // Push new annotations
+    pushUpdatedAnnotations(git)
+    pushLogger.info {"Done!"}
+}
+
+fun pushUpdatedAnnotations(git: Git) {
+
+    // Source: https://github.com/kittinunf/Kotlin-Playground/blob/master/buildSrc/src/main/java/com/github/kittinunf/gradle/ReleaseToMasterMergeTask.kt
+    class SshConfig(val sshKeyPath: String): TransportConfigCallback {
+        override fun configure(transport: Transport?) {
+            (transport as SshTransport).sshSessionFactory = object: JschConfigSessionFactory() {
+                override fun configure(hc: OpenSshConfig.Host?, session: Session?) {}
+                override fun createDefaultJSch(fs: FS?): JSch = super.createDefaultJSch(fs).apply {
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val status = git.status().call()
+
+    // A clean status indicates no updates
+    if (!status.isClean()) {
+        val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+        val formatter = SimpleDateFormat("yyyy-MM-dd-HH-mm-ss")
+
+        git.add().addFilepattern("data/").call()
+        git.commit()
+                .setMessage("includes annotations from ${currentDate.getTime()}").call()
+        git.tag().setName(formatter.format(currentDate.getTime())).call()
+        pushLogger.info {"Pushing updated data"}
+        git.push()
+                .setTransportConfigCallback(SshConfig("~/.ssh/id_rsa"))
+                .call()
+    } else {
+        pushLogger.info {"No updates - nothing will be pushed"}
+    }
+}
+
+fun setUpRepository(localWorkingCopyDirectory: String, repoToPushTo: String): Git {
+    /**
+     * Check if `localWorkingCopyDirectory` exists.
+     *   If it does: check the state
+     *   If not: clone `repoToPushTo` to the location specified by `localWorkingCopyDirectory`
+     */
+    if (File(localWorkingCopyDirectory).isDirectory()) {
+        val git = Git.open(File(localWorkingCopyDirectory))
```

Comment:
If you take a look at [the Javadoc for `Git`](https://download.eclipse.org/jgit/site/5.4.0.201906121030-r/apidocs/index.html) you'll see the warning that "The caller is responsible for closing the repository; close() on this instance does not close the repo.". Presumably it has some cleanup it needs to do when you are done with it.

Kotlin and recent versions of Java make this easier by providing an interface called `AutoCloseable` for things which can be automatically closed (include on exit via exceptions) using special language constructs (`try`-with-resources for Java, `use` for Kotlin; Python has a similar thing called "context managers").  

In this case, you aren't done with the `Git` object in the local scope, so what you have in this method is fine, but back in `main` you want to change it to something like:
```
setUpRepository(localWorkingCopyDirectory, repoToPushTo).use { git ->
    // Run ExportAnnotations.kt
    pushLogger.info {"Beginning ExportAnnotations"}
    ExportAnnotations.export(exportAnnotationsParams)
    // Run RestoreJson.kt
    if (restoreJson) {
        pushLogger.info { "Beginning RestoreJson" }
        RestoreJson.restore(restoreJsonParams)
    }

    // Push new annotations
    pushUpdatedAnnotations(git)
    pushLogger.info {"Done!"}
}
```

At the close of the `use` block `.close()` will automatically be called on the `Git` object.